source/sasl-auth.rst

```diff
@@ -0,0 +1,246 @@
+
+.. _windows-ad:
+
+Microsoft Windows Active Directory
+**********************************
+
+In order for LDAP operations to be performed
+against a Windows Active Directory server,
+a user record must be created to perform the lookups.
+
+The following example shows configuration parameters for ``saslauthd``
+to communicate with an Active Directory server:
+
+.. code-block:: text
+
+   ldap_servers: ldap://localhost
+   ldap_mech: PLAIN
+   ldap_search_base: CN=Users,DC=example,DC=com
+   ldap_filter: (sAMAccountName=%ucn=%u)
+   ldap_bind_dn: CN=ldapmgr,CN=Users,DC=<AD Domain>,DC=<AD TLD>
+   ldap_password: ld@pmgr_Pa55word
+
+In order to determine and test the correct search base
+and filter for your Active Directory installation,
+the Microsoft `LDP GUI Tool
+<https://technet.microsoft.com/en-us/library/Cc772839%28v=WS.10%29.aspx>`_
+can be used to bind and search the LDAP-compatible directory.
+
+
+4. Give write permissions to the :file:`/run/saslauthd` folder for the ``mongod``. Either change permissions to the  :file:`/run/saslauthd` folder:
+
+   .. code-block:: bash
+
+      $ sudo chmod 755 /run/saslauthd
+
+   Or add the ``mongod`` user to the ``sasl`` group:
+
+   .. code-block:: bash
+
+      $ sudo usermod -a -G sasl mongod
+
+Sanity check
+------------
+
+Verify that the ``saslauthd`` service can authenticate
+against the users created in the LDAP service:
+
+.. code-block:: bash
+
+   $ testsaslauthd -u christian -p secret  -f /var/run/saslauthd/mux
+
+This should return ``0:OK "Success"``.
+If it doesn't, then either the user name and password
+are not in the LDAP service, or ``sasaluthd`` is not configured properly.
+
+Configuring libsasl2
+--------------------
+
+The ``mongod`` also uses the SASL library for communications. To configure the SASL library, create a configuration file. 
+
+The configuration file **must** be named ``mongodb.conf`` and placed in a directory
+where ``libsasl2`` can find and read it.
+``libsasl2`` is hard-coded to look in certain directories at build time.
+This location may be different depending on the installation method.
+
+In the configuration file, specify the following:
+
+.. code-block:: text
+
+   pwcheck_method: saslauthd
+   saslauthd_path: /var/run/saslauthd/mux
+   log_level: 5
+   mech_list: plain
+
+The first two entries (``pwcheck_method`` and ``saslauthd_path``)
+are required for ``mongod`` to successfully use the ``saslauthd`` service.
+The ``log_level`` is optional but may help determine configuration errors.
+
+.. seealso::
+
+   `SASL documentation: <https://www.cyrusimap.org/sasl/index.html>`_
+  
+Configuring ``mongod`` Server
+-----------------------------
+
+To enable external authentication, you must create a user with the **root** privileges in the ``admin`` database. If you have already created this user, skip this step. Otherwise, run the following command to create the admin user:
```

Comment:
These is no need to create root user for external auth, however enabling access control (authorization: enabled) enforces authentication, so user is simply required for further actions with DB.

Reply:
So what role is sufficient for such a user?